msbuild/Xamarin.iOS.Tasks/Tasks/ILLinkBase.cs

```diff
@@ -24,11 +24,16 @@ public class ILLink : global::ILLink.Tasks.ILLink {
 		[Output]
 		public ITaskItem [] LinkedItems { get; set; } = Array.Empty<ITaskItem> ();
 
+		public string DotNetPath { get; set; } = string.Empty;
+
 		public override bool Execute ()
 		{
 			if (this.ShouldExecuteRemotely (SessionId))
 				return new TaskRunner (SessionId, BuildEngine4).RunAsync (this).Result;
 
+			if (!string.IsNullOrEmpty (DotNetPath))
```

Comment:
Should we do this in the Build agent or change anything else that's set there?

https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Messaging/Xamarin.Messaging.Build/TaskRunner.cs#L56-L85

Reply:
Yes, that might be a better solution, but it didn't seem trivial to copy the value of the _DotNetRootRemoteDirectory property from Windows to Mac, so I went with the easy solution. If you could let me know what needs to be done, , I can take a stab at it.

Reply:
Yes, that's a good point. I don't think we have an easy way to do that. 

If this is only intended for testing, what about using an env variable that's set on the bot and that the build agent will set as `DOTNET_HOST_PATH` if present. 

What is a bit odd of this implementation is part of the build will run with one dotnet host path, until it executes this task, and then it will continue with a different path. Is what you are setting different than what the Build agent is setting? Maybe it just works because no other tasks depend on it?